test/e2e/network_segmentation.go

```diff
@@ -1808,6 +1809,47 @@ spec:
 			),
 		)
 	})
+
+	Context("UDN with limited host subnet", func() {
+		It("check ovnkube-node pod restart when node is not assigned with UDN host subnet", func() {
+			nodes, err := e2enode.GetReadySchedulableNodes(context.TODO(), f.ClientSet)
+			framework.ExpectNoError(err)
+			if len(nodes.Items) < 3 {
+				ginkgo.Skip("requires at least 3 Nodes")
+			}
+
+			By("creating the network with limited host subnet configuration")
+			netConfig := networkAttachmentConfigParams{
+				name:     nadName,
+				topology: "layer3",
+				cidr:     correctCIDRFamily(cs, "192.168.100.0/25/26", "2014:100:200::/63/64"),
+				role:     "primary",
+			}
+			netConfig.namespace = f.Namespace.Name
+			udnManifest := generateUserDefinedNetworkManifest(&netConfig)
+			cleanup, err := createManifest(netConfig.namespace, udnManifest)
+			Expect(err).ShouldNot(HaveOccurred(), "creating udn network must succeed")
+			DeferCleanup(cleanup)
+			Eventually(userDefinedNetworkReadyFunc(f.DynamicClient, netConfig.namespace, netConfig.name), 5*time.Second, time.Second).Should(Succeed())
+
+			// Get the nodes again, it now updated with host subnet annotation
+			// for newly created UDN.
+			nodes, err = e2enode.GetReadySchedulableNodes(context.TODO(), cs)
+			framework.ExpectNoError(err)
+
+			restartpod := false
+			for i, node := range nodes.Items {
+				_, err = util.ParseNodeHostSubnetAnnotation(&nodes.Items[i], fmt.Sprintf("%s_%s", f.Namespace.Name, nadName))
+				if err != nil && util.IsAnnotationNotSetError(err) {
+					Expect(restartOVNKubeNodePod(cs, deploymentconfig.Get().OVNKubernetesNamespace(), node.Name)).ShouldNot(HaveOccurred(), "restart of OVNKube node pod must succeed")
+					restartpod = true
+					continue
+				}
+				Expect(err).ShouldNot(HaveOccurred(), "should not be other than host subnet annotation not set error")
+			}
+			Expect(restartpod).To(BeTrue())
```

Comment:
Here we're expecting to restart only 1 pod, right? So maybe `restartpod` should be an int that we expect to equal 1.

Reply:
there can be more than 3 nodes, so adding check now with `>= 1`.